azkaban-common/src/main/java/azkaban/trigger/JdbcTriggerImpl.java

```diff
@@ -0,0 +1,223 @@
+  private static final String GET_ALL_TRIGGERS =
+      "SELECT trigger_id, trigger_source, modify_time, enc_type, data FROM " + TRIGGER_TABLE_NAME;
+  private static final String GET_TRIGGER =
+      "SELECT trigger_id, trigger_source, modify_time, enc_type, data FROM " + TRIGGER_TABLE_NAME + " WHERE trigger_id=?";
+  private static final String ADD_TRIGGER = "INSERT INTO " + TRIGGER_TABLE_NAME + " ( modify_time) values (?)";
+  private static final String REMOVE_TRIGGER = "DELETE FROM " + TRIGGER_TABLE_NAME + " WHERE trigger_id=?";
+  private static final String UPDATE_TRIGGER =
+      "UPDATE " + TRIGGER_TABLE_NAME + " SET trigger_source=?, modify_time=?, enc_type=?, data=? WHERE trigger_id=?";
+  private static Logger logger = Logger.getLogger(JdbcTriggerImpl.class);
+  private final DatabaseOperator dbOperator;
+  private EncodingType defaultEncodingType = EncodingType.GZIP;
+
+  @Inject
+  public JdbcTriggerImpl(DatabaseOperator databaseOperator) {
+    this.dbOperator = databaseOperator;
+  }
+
+  @Override
+  public List<Trigger> getUpdatedTriggers(long lastUpdateTime) throws TriggerLoaderException {
+    logger.info("Loading triggers changed since " + new DateTime(lastUpdateTime).toString());
+
+    ResultSetHandler<List<Trigger>> handler = new TriggerResultHandler();
+
+    try {
+      List<Trigger> triggers = dbOperator.query(GET_UPDATED_TRIGGERS, handler, lastUpdateTime);
+      logger.info("Loaded " + triggers.size() + " triggers.");
+      return triggers;
+    } catch (SQLException ex) {
+      throw new TriggerLoaderException("Loading triggers from db failed.", ex);
+    }
+  }
+
+  @Override
+  public List<Trigger> loadTriggers() throws TriggerLoaderException {
+    logger.info("Loading all triggers from db.");
+
+    ResultSetHandler<List<Trigger>> handler = new TriggerResultHandler();
+
+    try {
+      List<Trigger> triggers = dbOperator.query(GET_ALL_TRIGGERS, handler);
+      logger.info("Loaded " + triggers.size() + " triggers.");
+      return triggers;
+    } catch (SQLException ex) {
+      throw new TriggerLoaderException("Loading triggers from db failed.", ex);
+    }
+  }
+
+  @Override
+  public void removeTrigger(Trigger t) throws TriggerLoaderException {
+    logger.info("Removing trigger " + t.toString() + " from db.");
+
+    try {
+      int removes = dbOperator.update(REMOVE_TRIGGER, t.getTriggerId());
+      if (removes == 0) {
+        throw new TriggerLoaderException("No trigger has been removed.");
+      }
+    } catch (SQLException ex) {
+      throw new TriggerLoaderException("Remove trigger " + t.getTriggerId() + " from db failed. ", ex);
+    }
+  }
+
+  /**
+   * TODO: Don't understand why we need synchronized here.
+   */
+  @Override
+  public synchronized void addTrigger(Trigger t) throws TriggerLoaderException {
+    logger.info("Inserting trigger " + t.toString() + " into db.");
+
+    SQLTransaction<Long> insertAndGetLastID = transOperator -> {
```

Comment:
This operation seems to contain only one update. Can it use the simpler update azkaban.db.DatabaseOperator#update method?